src/container/srv_container.c

```diff
@@ -1971,28 +1954,27 @@ cont_open(struct rdb_tx *tx, struct ds_pool_hdl *pool_hdl, struct cont *cont,
 		goto out;
 	}
 
-	/* lookup metadata times (NB: before updating open time in caller cont_op_with_svc() */
-	if (cont_proto_ver >= CONT_PROTO_VER_WITH_MDTIMES) {
-		struct co_md_times	mdtimes;
+	/* lookup metadata times (and pre-update open time to reflect this open operation.
+	 * NB client/engine may have recent (protocol) versions, however the pool may not
+	 * have been upgraded to new layout yet. mdtimes will be zeros in that case.
+	 */
+	rc = get_metadata_times(tx, cont, update_otime, false /* update_mtime */, &mdtimes);
+	if (rc != 0)
+		goto out;
 
-		/* NB client/engine may have recent (protocol) versions, however the pool may not
-		 * have been upgraded to new layout yet. mdtimes will be zeros in that case.
-		 */
-		rc = get_metadata_times(tx, cont, &mdtimes);
-		if (rc != 0)
-			goto out;
+	/* include metadata times in reply if client speaks the protocol */
+	if (mdtimes_in_reply && (opc_get(rpc->cr_opc) == CONT_OPEN)) {
+		struct cont_open_v7_out *out_v7 = crt_reply_get(rpc);
 
-		if (opc_get(rpc->cr_opc) == CONT_OPEN) {
-			struct cont_open_v7_out *out_v7 = crt_reply_get(rpc);
+		out_v7->coo_md_otime = mdtimes.otime;
+		out_v7->coo_md_mtime = mdtimes.mtime;
+	}
 
-			out_v7->coo_md_otime = mdtimes.otime;
-			out_v7->coo_md_mtime = mdtimes.mtime;
-		} else {	/* CONT_OPEN_BYLABEL */
-			struct cont_open_bylabel_v7_out *out_v7 = crt_reply_get(rpc);
+	if (mdtimes_in_reply && (opc_get(rpc->cr_opc) == CONT_OPEN_BYLABEL)) {
+		struct cont_open_v7_out *out_v7 = crt_reply_get(rpc);
```

Comment:
The type of out_v7 pointer variable declared here for CONT_OPEN_BYLABEL should have been struct cont_open_bylabel_v7_out *

This was the cause of the single test failure in build 9 - thought I had tested locally before pushing, but was mistaken. Fixing in a next/final push.